lib/msf/base/sessions/ssh_command_shell_bind.rb

```diff
@@ -82,7 +81,7 @@ def initialize(client, cid, ssh_channel, params)
         rsock.extend(SocketInterface)
         rsock.channel = self
 
-        lsock.extend(Rex::Socket::SslTcp) if params.ssl
+        lsock.extend(Rex::Socket::SslTcp) if params.ssl && !params.server
```

Comment:
To clarify is the idea here that if the parameter is SSL and the parameter also specifies this is a server that we should be using `Rex::Socket::SslTcpServer` here vs `Rex::Socket::SslTcp`? Just a little confused why we have this extra check here that wasn't here before.

Reply:
That's so the `lsock` is only extended with the client methods provided by `SslTcp` when creating a client socket since this code path is also executed when creating a server socket.

Reply:
Thanks for explaining, marking this as resolved as this makes sense.